src/main/java/homework8/ArrayListImplementation.java

```diff
@@ -0,0 +1,130 @@
+package homework8;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ArrayListImplementation<T> {
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private T[] typeArr;
+    private int size;
+
+    public ArrayListImplementation(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.typeArr = (T[]) new Object[initialCapacity];
+            this.size = 0;
+        } else if (initialCapacity == 0) {
+            this.typeArr =(T[]) EMPTY_ELEMENTDATA;
+            this.size = 0;
+        } else {
+            throw new IllegalArgumentException("Wrong capacity: " + initialCapacity);
+        }
+    }
+
+    private int size() {
+        return size;
+    }
+
+    private boolean isEmpty() {
+        return this.typeArr == EMPTY_ELEMENTDATA;
+    }
+
+    private void add(T value) {
+        if (isEmpty() || size == typeArr.length) {
```

Comment:
try to simplify:
if (isEmpty() || size == typeArr.length) {
            T[] newArr = (T[]) new Object[(typeArr.length * 3) / 2 + 1];
            System.arraycopy(typeArr, 0, newArr, 0, typeArr.length);
            typeArr = newArr;
        } 
            typeArr[size] = value;
        size++;